clouddrift/adapters/subsurface.py

```diff
@@ -0,0 +1,170 @@
+
+    # data
+    data_variables = ["dtnum", "lon", "lat", "p", "t", "u", "v"]
+    data = {}
+    for var in data_variables:
+        data[var] = np.concatenate([v.flatten() for v in source_data[var].flatten()])
+
+    # create rowsize variable
+    rowsize = np.array([len(v) for v in source_data["dtnum"].flatten()])
+    assert np.sum(rowsize) == len(data["dtnum"])
+
+    # some metadata are repeated for each float
+    # use those indices to retrieve one value per experiment
+    _, indices_exp = np.unique(metadata["indexExp"], return_index=True)
+
+    origin_datenum = 719529  # Unix epoch start (1970-01-01)
+
+    ds = xr.Dataset(
+        {
+            "expList": (["exp"], metadata["expList"]),
+            "expName": (["exp"], metadata["expName"][indices_exp]),
+            "expOrg": (["exp"], metadata["expOrg"][indices_exp]),
+            "expPI": (["exp"], metadata["expPI"][indices_exp]),
+            "indexExp": (["traj"], metadata["indexExp"]),
+            "fltType": (["traj"], metadata["fltType"]),
+            "id": (["traj"], metadata["indexFlt"]),
+            "rowsize": (["traj"], rowsize),
+            "time": (
+                ["obs"],
+                pd.to_datetime(data["dtnum"] - origin_datenum, unit="D"),
+            ),
+            "lon": (["obs"], data["lon"]),
+            "lat": (["obs"], data["lat"]),
+            "p": (["obs"], data["p"]),
+            "t": (["obs"], data["t"]),
+            "u": (["obs"], data["u"]),
+            "v": (["obs"], data["v"]),
+        }
+    )
+
+    # define attributes
+    vars_attrs = {
+        "expList": {
+            "long_name": "Experiment list",
+            "units": "-",
+        },
+        "expName": {
+            "long_name": "Experiment name",
+            "units": "-",
+        },
+        "expOrg": {
+            "long_name": "Experiment organization",
+            "units": "-",
+        },
+        "expPI": {
+            "long_name": "Experiment principal investigator",
+            "units": "-",
+        },
+        "indexExp": {
+            "long_name": "Experiment index number",
+            "units": "-",
+            "comment": "The index matches the float with its experiment metadata",
+        },
+        "fltType": {
+            "long_name": "Float type",
+            "units": "-",
+        },
+        "id": {"long_name": "Float ID", "units": "-"},
+        "lon": {"long_name": "Longitude", "units": "degrees_east"},
+        "lat": {"long_name": "Latitude", "units": "degrees_north"},
+        "rowsize": {
+            "long_name": "Number of observations per trajectory",
+            "sample_dimension": "obs",
+            "units": "-",
+        },
+        "u": {"long_name": "Eastward velocity", "units": "m/s"},
+        "v": {"long_name": "Northward velocity", "units": "m/s"},
+        "t": {"long_name": "Temperature", "units": "Celsius"},
+        "p": {"long_name": "Pressure", "units": "Millibar"},
```

Comment:
Do we care about CF standard names and units here? E.g. `sea_water_temperature` and `degree_C`. I think lon, lat, u, v are clear but at least "t" and "p" could be expanded to full names.

Reply:
Do we do this in other datasets? I don't see this in `gdp1h.py/gdp6h.py/mosaic.py` for example. I can rename the variables if this is preferred.

Reply:
I think it is for MOSAIC:

```python
    ds["longitude"].attrs = {
        "long_name": "longitude",
        "standard_name": "longitude",
        "units": "degrees_east",
    }

    ds["latitude"].attrs = {
        "long_name": "latitude",
        "standard_name": "latitude",
        "units": "degrees_north",
    }
```

but that one is easy because it only has longitude and latitude.

I don't know what's best to do. On one hand I think established conventions are good (our datasets are more likely to load correctly with NetCDF viewers for example; I don't know if that's a realistic use case), and on the other hand sometimes these standard names are unwieldy. I don't know if "sea_water_temperature" is necessary, but at the very least "temperature" (and perhaps "water_temperature") is more clear than "t". Yes, we see if from "long_name", but it helps to be able to see it from the variable name itself. Likewise for "p".

Reply:
I put `temp` and `pres` but added the correct corresponding `standard_name`.. kinda following what I see here, https://cf-xarray.readthedocs.io/en/latest/examples/introduction.html. As long as the variables are _readable_, and we assigned the corresponding `"standard_name"` and `"units"`.

<img width="835" alt="image" src="https://github.com/Cloud-Drift/clouddrift/assets/387624/2459ea6c-923d-478c-a6b7-432c9811331e">

Reply:
I think I fixed the issues with the units too. Let me know if you see anything else!